### PR TITLE
Transform to multi create

### PR DESCRIPTION
**Pull request by sasurobert**
Transform to multi create

### CODE REVIEW

File: vm/systemSmartContracts/esdt.go [hunk 1]
```diff
@@ -1326,6 +1332,66 @@ func (e *esdt) checkSpecialRolesAccordingToTokenType(args [][]byte, token *ESDTD
...
+	}
+
+	addressWithCreateRole, err := getFirstAddressWithGivenRole(token, []byte(core.ESDTRoleNFTCreate))
+	if err != nil {
+		e.eei.AddReturnMessage(err.Error())
+		return vmcommon.UserError
+	}
+
+	token.CanCreateMultiShard = true
+	isAddressLastByteZero := addressWithCreateRole[len(addressWithCreateRole)-1] == 0
```
Comment (iulianpascalau) [inline vm/systemSmartContracts/esdt.go:1376]: why do we need this?

File: vm/systemSmartContracts/esdt.go [hunk 2]
```diff
@@ -1326,6 +1332,66 @@ func (e *esdt) checkSpecialRolesAccordingToTokenType(args [][]byte, token *ESDTD
...
+	}
+	if !token.CanAddSpecialRoles {
+		e.eei.AddReturnMessage("cannot add special roles")
+		return vmcommon.UserError
+	}
+	if token.CanCreateMultiShard {
+		e.eei.AddReturnMessage("it is already multi shard create")
+		return vmcommon.UserError
+	}
+
```
Comment (iulianpascalau) [inline vm/systemSmartContracts/esdt.go:1368]: no eei.UseGas ?

File: node/nodeRunner.go [hunk 3]
```diff
@@ -150,7 +150,6 @@ func printEnableEpochs(configs *config.Configs) {
 	log.Debug(readEpochFor("esdt NFT create on multiple shards"), "epoch", enableEpochs.ESDTNFTCreateOnMultiShardEnableEpoch)
 	log.Debug(readEpochFor("SCR size invariant check"), "epoch", enableEpochs.SCRSizeInvariantCheckEnableEpoch)
 	log.Debug(readEpochFor("backward compatibility flag for save key value"), "epoch", enableEpochs.BackwardCompSaveKeyValueEnableEpoch)
-	log.Debug(readEpochFor("esdt NFT create on multiple shards"), "epoch", enableEpochs.ESDTNFTCreateOnMultiShardEnableEpoch)
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply node/nodeRunner.go:153]: It is duplicated.

File: vm/systemSmartContracts/esdt.go (hunk 1 above)
Comment (sasurobert) [reply vm/systemSmartContracts/esdt.go:1376]: You do not need to put a new start nonce for the address which ends in zero. That address already made NFTs from 0 to X.

File: vm/systemSmartContracts/esdt.go (hunk 2 above)
Comment (sasurobert) [reply vm/systemSmartContracts/esdt.go:1368]: basicOwnershipChecks has inside a useGas

File: node/nodeRunner.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply node/nodeRunner.go:153]: ok

**Review by iulianpascalau — APPROVED** [review]

**Review by dragos-rebegea — APPROVED** [review]

**Pull request merged**